### PR TITLE
[Refactor] JavaScript worker tests

### LINKED ISSUE #228: [Refactor] JavaScript worker tests
Now tests of JS worker have a shitty MongoDB mock. It needs to refactor this tests using `mongodb-memory-server`.

### PR DESCRIPTION
**Pull request by ilyamore88**
I refactored JS worker tests. Now it doesn't use MongoDB mock.
I implemented the test for source maps parsing but didn't realize it. I will do it in the next pull request for issue #226.
Closes #228

### CODE REVIEW

File: workers/javascript/tests/index.test.ts [hunk 1]
```diff
@@ -1,102 +1,276 @@
...
+
+  beforeAll(async () => {
+    connection = await MongoClient.connect(process.env.MONGO_EVENTS_DATABASE_URI, {
+      useNewUrlParser: true,
+      useUnifiedTopology: true,
+    });
+    db = connection.db('hawk');
+  });
+
+  it('should have correct catcher type', () => {
```
Comment (neSpecc) [inline workers/javascript/tests/index.test.ts:103]: Useless test

File: workers/javascript/tests/index.test.ts [hunk 2]
```diff
@@ -1,102 +1,276 @@
...
+     */
+    const workerType = worker.type;
+
+    /**
+     * Assert
+     */
+    expect(workerType).toEqual('errors/javascript');
+  });
+
+  it('should start correctly', async () => {
```
Comment (neSpecc) [inline workers/javascript/tests/index.test.ts:120]: Useless test

File: workers/javascript/tests/index.test.ts [hunk 3]
```diff
@@ -1,102 +1,276 @@
...
+    /**
+     * Assert
+     *
+     * No errors
+     */
+    await worker.finish();
   });
 
-  test('should start correctly', async () => {
+  it('should finish correctly', async () => {
```
Comment (neSpecc) [inline workers/javascript/tests/index.test.ts:141]: Useless test

File: workers/javascript/tests/index.test.ts [hunk 4]
```diff
@@ -1,102 +1,276 @@
...
+    /**
+     * Assert
+     *
+     * No errors
+     */
   });
 
-  test('should handle right messages', async () => {
-    await worker.handle(testEventData);
+  it('should handle event and add task to grouper', async () => {
```
Comment (neSpecc) [inline workers/javascript/tests/index.test.ts:163]: what does it mean 'handle'? Please add more behavior-oriented specification

File: workers/javascript/tests/index.test.ts [hunk 5]
```diff
@@ -1,102 +1,276 @@
...
+        title: 'Mocked event for JS event worker',
+        type: 'Error',
+        timestamp: Date.now(),
+        release: '3fa0f290c014',
+        addons: {
+          window: {
+            innerHeight: 1337,
+            innerWidth: 960,
+          },
+          userAgent: withUserAgent && 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:80.0) Gecko/20100101 Firefox/80.0',
```
Comment (neSpecc) [inline workers/javascript/tests/index.test.ts:49]: I'd suggest moving it to the `originalUserAgent` variable and place it near the `beautifiedUserAgend`

**Review by neSpecc — CHANGES_REQUESTED** [review]

**Review by neSpecc — APPROVED** [review]

**Review by talyguryn — APPROVED** [review]

**Pull request merged**